app/controllers/concerns/active_model_serializers_fix.rb

```diff
@@ -0,0 +1,10 @@
+module ActiveModelSerializersFix
+  def namespace_for_serializer
+    @namespace_for_serializer ||=
+      if Module.method_defined?(:parent)
+        self.class.parent unless self.class.parent == Object
+      else
+        self.class.module_parent unless self.class.module_parent == Object
+      end
+  end
+end
```

Comment:
```suggestion
module ActiveModelSerializersFix
  if Module.method_defined?(:parent)
    def namespace_for_serializer
      @namespace_for_serializer ||= self.class.parent unless self.class.parent == Object
    end
  else
    def namespace_for_serializer
      @namespace_for_serializer ||= self.class.module_parent unless self.class.module_parent == Object
    end
  end
end
```

Small improvement to avoid wasting runtime proc time.